### ISSUE TITLE
Fail to build tokio docs on nightly

### ISSUE THREAD

**Issue by kpp** [rust-lang/rust#56693]
https://github.com/rust-lang/rust/blob/1137d29d5e551e377579c5a601fe7c444057d00c/src/libstd/io/error.rs#L215

This error lives on nightly for ages:

```
$ rustc -Vv
rustc 1.32.0-nightly (4a45578bc 2018-12-07)
binary: rustc
commit-hash: 4a45578bc58ff262864f72680cc02e83f5d2f5b3
commit-date: 2018-12-07
host: x86_64-unknown-linux-gnu
release: 1.32.0-nightly
LLVM version: 8.0
```

```
~/tokio$ cargo doc
 Documenting tokio v0.1.13 (/home/humbug/tokio)
error: `[ErrorKind]` cannot be resolved, ignoring it...
    |
note: lint level defined here
   --> src/lib.rs:2:23
    |
2   | #![deny(missing_docs, warnings, missing_debug_implementations)]
    |                       ^^^^^^^^
    = note: #[deny(intra_doc_link_resolution_failure)] implied by #[deny(warnings)]
    = note: the link appears in this line:
            
             Converts an [`ErrorKind`] into an [`Error`].
                          ^^^^^^^^^^^
    = help: to escape `[` and `]` characters, just add '\' before them like `\[` or `\]`

error: `[Error]` cannot be resolved, ignoring it...
    |
    = note: the link appears in this line:
            
             Converts an [`ErrorKind`] into an [`Error`].
                                                ^^^^^^^
    = help: to escape `[` and `]` characters, just add '\' before them like `\[` or `\]`

error: Could not document `tokio`.

Caused by:
  process didn't exit successfully: `rustdoc --crate-name tokio src/lib.rs --color always -o ..............
```

I am not sure this is an std::io::Error error, but it looks very strange there is no any link to FILE:LINE.

The source code can be found at https://github.com/tokio-rs/tokio/pull/792

**Comment by kpp** [comment]
> Update the linked doc with the following two lines:

Hold on. The bug would be found in 10 seconds if you provided FILE:LINE. So the issue is a little bit more complicated.

**Comment by euclio** [comment]
Shouldn't this warning be capped when documenting tokio since it occurs in the standard library?

cc #51684

**Comment by brunoczim** [comment]
This is why `#[deny(warnings)]` is a really bad idea.

**Comment by euclio** [comment]
Minimal repro:

```
pub use std::io::Error;
```

**Comment by kpp** [comment]
Still not fixed in `rustc 1.33.0-nightly (2d3e909e4 2018-12-22)`.

**Comment by kpp** [comment]
Fixed <3
